stacks-signer/src/signerdb.rs

```diff
@@ -976,6 +1011,73 @@ impl SignerDb {
         ))
     }
 
+    /// Get a pending block validation, sorted by the time at which it was added to the pending table.
+    /// If found, remove it from the pending table.
+    pub fn get_and_remove_pending_block_validation(
+        &self,
+    ) -> Result<Option<Sha512Trunc256Sum>, DBError> {
+        if let Some(sighash) = self.get_pending_block_validation()? {
+            self.remove_pending_block_validation(&sighash)?;
+            return Ok(Some(sighash));
+        }
```

Comment:
Would it be more efficient to do the get and delete in one query, like a `DELETE` with a `RETURNING signer_signature_hash`?